Implement buffer filling and sequence fulfillment

Ablu opened the pull request.

Ablu committed:
Implement buffer filling and sequence fulfillment

Ablu committed:
Allow winning or losing

Kugelschieber commented on src/game/Game.test.ts:
```diff
@@ -38,11 +39,66 @@ describe("GameState", () => {
+            game.pick(1, 0)
+            game.pick(1, 2)
+            expect(game.state).toEqual(EndState.Lost)
+        })
```
Test to make sure pick does not work after the buffer is full.

Ablu replied:
done

Ablu committed:
Time management

The game now fails after a configurable amount of time. Winning pauses
the clock.

Ablu committed:
Added return type to getCell()
